src/react-native/04-intro-to-jsx/intro-to-jsx.md

```diff
@@ -10,17 +10,177 @@
 
 In this section, you will:
 
-- TODO
+- Learn the basics of JSX
+- Discover the differences between JSX and HTML
+- Use JavaScript variables and expressions in JSX
+- Work with conditionals and loops in JSX
 
 ## Objective 1: Create a UI with JSX
 
 Now that we have our project set up, let’s update our page to look like the design below:
 
 <img alt="Screenshot of a mobile application interface text “Place My Order: Coming Soon To...” and “Illinois“." src="../../static/img/react-native/04-intro-to-jsx/01-solution.png" style="max-height: 750px; border: 4px solid black; border-radius: 25px;"/>
 
-### Concept TODO
+### What is JSX?
 
-TODO
+JSX is used by React Native developers to define the user interface.
+
+JSX is a special syntax designed to look almost identical to HTML. Developers define the UI using JSX. React transforms that JSX into the HTML the browser displays. JSX can be written alongside standard JavaScript, which makes for a powerful programming environment.
+
+#### Basic JSX looks like HTML
+
+```tsx
+const greeting = <Text>Hello, world!</Text>
+```
+
+This code snippet creates a simple JSX element: a `<Text>` header with the text “Hello, world!” This is similar to writing HTML, but it’s JSX inside a JavaScript file.
+
+### Embedding a JavaScript expression in JSX
+
+```tsx
+const name = "Alice"
+const greeting = <Text>Hello, {name}!</Text>
+```
+
+Here, we embed a JavaScript expression `{name}` within JSX. The value of the name variable is displayed within the `<Text>` tag.
+
+### Combining JSX with standard JavaScript
+
+```tsx
+function welcome(name) {
+  return <Text>Hello, {name}</Text>
+}
+
+const welcomeMessage = welcome("Alice")
+```
+
+This example illustrates a regular JavaScript function being called with a single argument. The `welcome` function returns JSX, showing how JSX can be seamlessly integrated within standard JavaScript functions.
+
+### JSX transpiled
+
+React Native has a procedural `React.createElement` syntax, but most applications do not use it directly. Instead, views are defined and maintained in JSX and will automatically be transpiled into the equivalent `React.createElement` calls at build-time.
+
+```tsx
+function welcome(name) {
+  return React.createElement("Text", null, `Hello, ${name}!`)
+}
+
+const welcomeMessage = welcome("Alice")
+```
+
+This transformation is handled by tools like Vite during the build process, allowing us to write more readable and maintainable code using JSX.
+
+### Differences between JSX and HTML
+
+JSX looks a lot like HTML, and that’s intentional. Browsers understand HTML, JavaScript, and CSS. Eventually, anything we build has to be converted into one of those 3 syntaxes. Since our UI code will eventually end up as HTML, using JSX means it will be easier to understand and debug the end result in the browser.
+
+However, JSX is converted into JavaScript and therefore is not an exact mirror of HTML. Some of the most noticeable differences include:
```

Comment:
While JSX syntactically resembles XML or HTML, there are several key differences in how they are used and processed. Understanding these differences is crucial for developers transitioning from XML’s strict standards or HTML’s looser structure.

Here are some of the most noticeable differences between JSX and XML: